### PR TITLE
Install bot2-frames and bot2-param

### PR DESCRIPTION
**Pull request by jamiesnape**
Toward #3129. This should now be sufficient for Director et al.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/6358)

### CODE REVIEW

**Comment by jamiesnape** [comment]
+@mwoehlke-kitware for feature review.

**Comment by mwoehlke-kitware** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-Kmlu-_ALKGTqC-1m9nS:b-i0iol3) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
> ```
> ]
> 
> lcm_c_aggregate_header(
> ```

Any reason to not just lump this into `lcm_c_library`?

```
lcm_c_library(
  ...
  aggregate_hdr = 1,
  ...
)
```

---

*[tools/libbot.BUILD, line 191 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmlubGm3dwOdDs8_6Cw:-KmlubGm3dwOdDs8_6Cx:b-myk09j) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L191)):*
> ```
> )
> 
> BOT2_PARAM_PUBLIC_HDRS = [
> ```

Maybe let's add a TODO to create a helper function to compute these from the LCM inputs? (Which the LCM utility functions would also use...)

---

*[tools/libbot.BUILD, line 207 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlut9W-2MI6_NEOnPa:-Kmlut9W-2MI6_NEOnPb:bb6u109) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L207)):*
> ```
>         # The following need to be in both src and hdrs because include_prefix
>         # is being used.
>         "bot2-param/src/param_client/param_client.h",
> ```

DRY?

---

*[tools/libbot.BUILD, line 326 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmlvDo6qJXlfJeYKzRG:-KmlvDo7asBMZ18ansJx:bb6u109) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L326)):*
> ```
>         # The following needs to be in both src and hdrs because include_prefix
>         # is being used.
>         "bot2-frames/src/bot_frames.h",
> ```

DRY?

---

*[tools/libbot.BUILD, line 364 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmlvRY2nCPFliLngqq2:-KmlvRY2nCPFliLngqq3:bdhbjwj) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L364)):*
> ```
>     guess_hdrs = "PACKAGE",
>     hdr_strip_prefix = [
>         "bot2-frames",
> ```

FYI, you _probably_ could use `"*"` here, though not sure that's preferable...

---

*[tools/libbot.BUILD, line 367 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlx5Id1RDDkrv7CIc5:-Kmlx5Id1RDDkrv7CIc6:b-q7pef8) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L367)):*
> ```
>         "bot2-param",
>     ],
>     license_docs = LICENSE_DOCS,
> ```

Sigh. This _really_ should only be installed once... I don't think we should be enforcing that every install must install licenses; instead, we should check somewhere that externals have installed licenses. (Not in this PR, but at some point...)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by jamiesnape** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmFuZWSy5UGuzttsd1:b-9vxsvy) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Any reason to not just lump this into `lcm_c_library`?

```
lcm_c_library(
  ...
  aggregate_hdr = 1,
  ...
)
```
</blockquote></details>

The aggregate header name is actually arbitrary, and not sure I want to be encouraging these headers anyway.

---

*[tools/libbot.BUILD, line 191 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmlubGm3dwOdDs8_6Cw:-KmmHb8rxWhnxEKzT6Lh:b-gb4ag6) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L191)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Maybe let's add a TODO to create a helper function to compute these from the LCM inputs? (Which the LCM utility functions would also use...)
</blockquote></details>

These are not related to the LCM inputs, so that would not make sense.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by mwoehlke-kitware** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmIJIViD32pXSlZjte:b-rowdzf) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

The aggregate header name is actually arbitrary, and not sure I want to be encouraging these headers anyway.
</blockquote></details>

The _rule_ name is somewhat arbitrary... so what? The name of the header itself is not; the header is `'%(lcm_package)s.h'`. (This is enforced in CMake; I see no reason Bazel shouldn't do likewise.)

---

*[tools/libbot.BUILD, line 191 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmlubGm3dwOdDs8_6Cw:-KmmIjRvJRK0i9DpB9Xl:bhneh9i) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L191)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

These are not related to the LCM inputs, so that would not make sense.
</blockquote></details>

Right... NM...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by jamiesnape** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmLZex9F4fpD78tM9O:b-rv0got) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

The _rule_ name is somewhat arbitrary... so what? The name of the header itself is not; the header is `'%(lcm_package)s.h'`. (This is enforced in CMake; I see no reason Bazel shouldn't do likewise.)
</blockquote></details>

No, it is actually PROJECT_NAME with dashes changed to underscores.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by jamiesnape** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmLkOlHTa3cJ5zDESC:b8qez9b) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

No, it is actually PROJECT_NAME with dashes changed to underscores.
</blockquote></details>

Hence bot2_param and bot2_frames. That is overridden for bot_core.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by jamiesnape** [comment]
Review status: 4 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmQDhJo-lx8heBATWS:b-rss2zk) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Hence bot2_param and bot2_frames. That is overridden for bot_core.
</blockquote></details>

Basically, if I "fix" this, I break compilation of libbot.

---

*[tools/libbot.BUILD, line 207 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlut9W-2MI6_NEOnPa:-KmmQCWIvgIyaxOxf6H0:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L207)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

DRY?
</blockquote></details>

Done.

---

*[tools/libbot.BUILD, line 326 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmlvDo6qJXlfJeYKzRG:-KmmQCw442fyM3fdKnS9:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/libbot.BUILD#L326)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

DRY?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by mwoehlke-kitware** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmQqSzAZDQY1Xjc6yr:b-goa67v) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Basically, if I "fix" this, I break compilation of libbot.
</blockquote></details>

Hmm... then [`lcm_wrap_types`](https://github.com/lcm-proj/lcm/blob/master/lcm-cmake/lcmUtilities.cmake#L157) must also be "broken"?

(IMNSHO, the **correct** name for the header **is** `'%(lcm_package)s.h'`; if libbot is naming it something else, _libbot_ is broken.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by jamiesnape** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmRomaAINmJk-xMdKD:bgegye1) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Hmm... then [`lcm_wrap_types`](https://github.com/lcm-proj/lcm/blob/master/lcm-cmake/lcmUtilities.cmake#L157) must also be "broken"?

(IMNSHO, the **correct** name for the header **is** `'%(lcm_package)s.h'`; if libbot is naming it something else, _libbot_ is broken.)
</blockquote></details>

Libbot uses its own logic: https://github.com/RobotLocomotion/libbot2/blob/master/bot2-frames/cmake/lcmtypes.cmake and https://github.com/RobotLocomotion/libbot2/blob/master/bot2-param/cmake/lcmtypes.cmake.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by mwoehlke-kitware** [comment]
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmT8iPDle-qoHtMZmv:bzg5w2o) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Libbot uses its own logic: https://github.com/RobotLocomotion/libbot2/blob/master/bot2-frames/cmake/lcmtypes.cmake and https://github.com/RobotLocomotion/libbot2/blob/master/bot2-param/cmake/lcmtypes.cmake.
</blockquote></details>

Well, yes, it predates CMake-ified LCM. I'll still assert that it (libbot) is broken :smile:.

Hmm... make it a TODO for now?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by mwoehlke-kitware** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/bot_core_lcmtypes.BUILD, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-Kmlu-_9Eb6JbdNn2C3s:-KmmTSMuBrhizGDz3KQA:bpkzwho) ([raw file](https://github.com/robotlocomotion/drake/blob/d27c42ead91cf631b80353e8d96ac85b07e68558/tools/bot_core_lcmtypes.BUILD#L28)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Well, yes, it predates CMake-ified LCM. I'll still assert that it (libbot) is broken :smile:.

Hmm... make it a TODO for now?
</blockquote></details>

...which I see you just did: :+1:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358)*

**Comment by mwoehlke-kitware** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tools/lcm.bzl, line 255 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmmTfFXY95vDOhyUJc0:-KmmTfFXY95vDOhyUJc1:brzhtr8) ([raw file](https://github.com/robotlocomotion/drake/blob/101a324ff2912e0b211c3c00f5d54e7537c9cc63/tools/lcm.bzl#L255)):*
> ```Python
>         **kwargs)
> 
> # TODO(jamiesnape): Simplify this and possibly merge with lcm_c_library if
> ```

FYI, feel free to put my name here instead...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-:-KmmTmpoMmMrpmdodlej:bnfp4nl)*

**Comment by jamiesnape** [comment]
Nominally +@jwnimmer-tri for platform review. Please reassign as appropriate.

**Comment by jwnimmer-tri** [comment]
:lgtm:

---

Reviewed 3 of 5 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[tools/libbot.BUILD, line 209 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmmmaRYx7AmPmiD_ocO:-KmmmaRYx7AmPmiD_ocP:bbfk49w) ([raw file](https://github.com/robotlocomotion/drake/blob/101a324ff2912e0b211c3c00f5d54e7537c9cc63/tools/libbot.BUILD#L209)):*
> ```
>     ],
>     hdrs = BOT2_PARAM_PUBLIC_HDRS,
>     copts = ["-std=gnu99"],
> ```

FYI Is this a bot2-wide setting?  (Or is it also declared in multiple locations upstream?)  If its generally applicable, consider factoring it our or using a named constant here.

---

*[tools/robotlocomotion_lcmtypes.BUILD, line 25 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-KmmlkGuSU11QZEm8kOj:-KmmlkGv8_x78UECYO6L:b-3u5vys) ([raw file](https://github.com/robotlocomotion/drake/blob/101a324ff2912e0b211c3c00f5d54e7537c9cc63/tools/robotlocomotion_lcmtypes.BUILD#L25)):*
> ```
>     name = "robotlocomotion_lcmtypes_c_aggregate_header",
>     out = "lcmtypes/robotlocomotion.h",
>     lcm_package = "robotlocomotion",
> ```

FYI This this repeated in several places here and below; a named constant might be better?  (Future PR is OK.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6358#-:-KmmnFXd1R_lWLko6YGd:bnfp4nl)*
